src/Lecture1.hs

```diff
@@ -119,4 +142,34 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+
+-- Simpler implementation with let and straiforward functions:
+-- lowerAndGreater :: Int -> [Int] -> [Char]
+-- lowerAndGreater n list = 
+--     let calculateMin x acc l = if null l then acc else calculateMin x (if (head l) < x then (acc + 1) else acc) (tail l)
+--         calculateMax x acc l = if null l then acc else calculateMax x (if (head l) > x then (acc + 1) else acc) (tail l)
+--     in show n ++ " is greater than " ++ show (calculateMin n 0 list) ++ " elements and lower than " ++ show (calculateMax n 0 list) ++ " elements"
+
+-- Implementation with where clause and operator passing as an argument:
+-- lowerAndGreater :: Int -> [Int] -> [Char]
+-- lowerAndGreater n list = show n ++ " is greater than " ++ show (calculateMin n 0 list) ++ " elements and lower than " ++ show (calculateMax n 0 list) ++ " elements"
+--     where 
+--         calculate :: Int -> Int -> [Int] -> (Int -> Int -> Bool) -> Int
+--         calculate x acc l op = if null l then acc else (calculate x (if (op (head l) x) then (acc + 1) else acc) (tail l) op)
+--         calculateMin :: Int -> Int -> [Int] -> Int
+--         calculateMin x acc l = calculate x acc l (>)
+--         calculateMax :: Int -> Int -> [Int] -> Int
+--         calculateMax x acc l = calculate x acc l (<)
+
+-- Implementation with single list traversal
+lowerAndGreater :: Int -> [Int] -> [Char]
+lowerAndGreater n list = show n ++ " is greater than " ++ show mn ++ " elements and lower than " ++ show mx ++ " elements"
+    where 
+        getMinMax :: Int -> Int -> Int -> [Int] -> (Int, Int)
+        getMinMax x accMin accMax l 
+            | null l = (accMin, accMax) 
+            | (head l) > x = getMinMax x accMin (accMax + 1) (tail l)
+            | (head l) < x = getMinMax x (accMin + 1) accMax (tail l)
+            | otherwise = getMinMax x accMin accMax (tail l)
```

Comment:
Great usage of guards! 💂🏻 
This new solution looks really neat 💯 

I haven't explained tuples yet, as you correctly noticed. So you're already one step ahead! 🦶🏻 
One way to solve this problem without using tuples is to make it return `[Char]`. So you construct the string in the `null l` guard at the end of recursion and you need to create a temporary tuple. But I like the solution with a pair as it separates counting logic from formatting.

One last nitpick is that you don't need to pass `n` to the `getMinMax` function initially and you can get rid of the `x` argument. You can notice that it's always `x` (i.e. `n`) and it doesn't change during recursive calls. Functions inside `where` can see all the arguments of the main function, so you can just remove the extra `x` argument in recursive calls and simply compare with `n`.

Reply:
Wow,! Indeed, I can see the whole redundancy of the x now :) Thank you very much for your feedback and comments!